custom_components/dreo/pydreo/pydreoheater.py

```diff
@@ -0,0 +1,414 @@
+"""Dreo API for controling heaters."""
+
+import logging
+from typing import TYPE_CHECKING, Dict
+
+from .constant import (
+    LOGGER_NAME,
+    HTALEVEL_KEY,
+    TEMPERATURE_KEY,
+    MODE_KEY,
+    OSCON_KEY,
+    MUTEON_KEY,
+    POWERON_KEY,
+    DEVON_KEY,
+    TIMERON_KEY,
+    COOLDOWN_KEY,
+    PTCON_KEY,
+    LIGHTON_KEY,
+    CTLSTATUS_KEY,
+    TIMEROFF_KEY,
+    ECOLEVEL_KEY,
+    CHILDLOCKON_KEY,
+    TEMPOFFSET_KEY,
+    FIXEDCONF_KEY,
+    HEATER_MODE_COOLAIR,
+    HEATER_MODE_HOTAIR,
+    HEATER_MODE_ECO,
+    HEATER_MODE_OFF,
+    HEATER_MODES,
+    MODE_LEVEL_MAP,
+    LEVEL_MODE_MAP,
+    TemperatureUnit
+)
+
+from homeassistant.components.climate import (
+    FAN_ON, 
+    FAN_OFF
+)
+
+from .pydreobasedevice import PyDreoBaseDevice
+from .models import DreoDeviceDetails, HEAT_RANGE, ECOLEVEL_RANGE
+
+_LOGGER = logging.getLogger(LOGGER_NAME)
+
+if TYPE_CHECKING:
+    from pydreo import PyDreo
+
+class PyDreoHeater(PyDreoBaseDevice):
+    """Base class for Dreo heater API Calls."""
+
+    def __init__(self, device_definition: DreoDeviceDetails, details: Dict[str, list], dreo: "PyDreo"):
```

Comment:
How much of this is duplicative of what's in Fan?

Reply:
The structure is the same, but the methods and properties are largely different. There are some common things that should at some point be refactored out into the base device. But as I noted, I wanted to keep things as similar to the old code as possible to avoid breaking the fans because I don't have one and can't easily test it.